src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,148 @@
+    setIsLoaded();
+  }
+
+  const sortedBy = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value: typeOfSort } = event.target;
+    setSelectedSort(typeOfSort);
+  }
+
+  const sortedTodos = useMemo(() => {
+    return selectedSort === 'initial'
+        ? [...todos]
+        : [...todos].sort((aTodo, bTodo) => {
+          switch(selectedSort) {
+            case 'title':
+              return aTodo.title.localeCompare(bTodo.title);
+            case 'name':
+              return aTodo.user && bTodo.user ? aTodo.user.name.localeCompare(bTodo.user.name) : 0;
+            case 'completed':
+              return Number(aTodo.completed) - Number(bTodo.completed);
+            default:
+            return 0;
+          }
+        })
+
+  }, [todos, selectedSort]);
+
+  if (isLoading) {
+    return (
+      <p className="loading">
+        Loading...
+      </p>
+    );
+  }
+
+  return (
+    <div className="app">
+      <h1 className="title">Dynamic list of TODOs</h1>
+      {(!isLoaded ? (
+          <>
+            <p className="initual_loading">
+              Load your Todos
+            </p>
+            <button
+              type="button"
+              className="loading_button"
+              onClick={showAllTodos}
+            >
+              Load
+            </button>
+          </>
+         )
+        : (
+          <>
+            <select value={selectedSort}
+              onChange={sortedBy}
+              className="selected_button"
+            >
+              <option disabled value='choose'>Choose sort method</option>
+              <option value='initial'>Initial view</option>
+              <option value='title'>Title</option>
+              <option value='name'>Name</option>
+              <option value='completed'>Completed</option>
+            </select>
+            <table className="table">
+              <thead>
+                <tr className="table__title">
+                  <th className="table__cell">User</th>
+                  <th className="table__cell">Todo</th>
+                  <th className="table__cell">Status</th>
+                  <th className="table__cell">Delete</th>
+                </tr>
+              </thead>
+              <tbody>
+              <>
+                {sortedTodos.map(todo => <Todo key={todo.id} todo={todo} />)}
+              </>
+              </tbody>
+            </table>
+          </>
+        )
+      )}
+    </div>
+  )
+}
+
+const mapStateToProps = (state: State) => ({
+  isLoading: state.isLoading,
+  isLoaded: state.isLoaded,
+  todos: state.todos,
+  selectedSort: state.selectedSort
+});
+const mapDispatchToProps = (dispatch: Dispatch) => {
+  return {
+    setIsLoading: (status: boolean) => dispatch(setIsLoading(status)),
+    setIsLoaded: () => dispatch(setIsLoaded()),
+    setTodos: (todosFromApi: PreparedTodo[]) => dispatch(setTodos(todosFromApi)),
+    setSelectedSort: (sort: string) => dispatch(setSelectedSort(sort))
+  }
+};
```

Comment:
Редакс прокинет пропсы и все остальное сделает за тебя
```suggestion
});

const mapDispatchToProps = {
    setIsLoading,
    setIsLoaded,
    setTodos,
    setSelectedSort,
};
```